domain/src/Commands/DomainCommands.php

```diff
@@ -0,0 +1,1288 @@
+  }
+
+  /**
+   * Validates a domain.
+   *
+   * @param $hostname
+   *   The domain name to validate for syntax and uniqueness.
+   *
+   * @return array
+   *   An array of errors encountered.
+   *
+   * @see domain_validate()
+   */
+  protected function validateDomain($hostname) {
+    /** @var \Drupal\domain\DomainValidatorInterface $validator */
+    $validator = \Drupal::service('domain.validator');
+    return $validator->validate($hostname);
+  }
+
+  /**
+   * Deletes a domain record.
+   *
+   * @param DomainInterface[] $domains
+   *   The domain_id to delete. Pass 'all' to delete all records.
+   *
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   * @throws \UnexpectedValueException
+   */
+  protected function deleteDomain($domains) {
+    foreach ($domains as $domain) {
+      if (!$domain instanceof DomainInterface) {
+        throw new StorageException('deleting domains: value is not a domain');
+      }
+      $hostname = $domain->getHostname();
+
+      if ($this->is_dry_run) {
+        $this->logger()->info(dt('DRYRUN: Domain record @domain deleted.',
+          ['@domain' => $hostname]));
+        continue;
+      }
+
+      try {
+        $domain->delete();
+      }
+      catch (EntityStorageException $e) {
+        throw new DomainCommandException(dt('Unable to delete domain: @domain',
+          ['@domain' => $hostname]), $e);
+      }
+      $this->logger()->info(dt('Domain record @domain deleted.',
+        ['@domain' => $hostname]));
+    }
+  }
+
+  /**
+   * Determine a list of the entities that are domain enabled.
+   *
+   * @return string[]
+   *   List of entities supporting domain access.
+   */
+  protected function findDomainEnabledEntities(string $using_field = DOMAIN_ACCESS_FIELD) {
+    $this->ensureEntityFieldMap();
+    $entities = [];
+    foreach($this->entity_field_map as $type => $fields) {
+      if (array_key_exists($using_field, $fields)) {
+        $entities[] = $type;
+      }
+    }
+    return $entities;
+  }
+
+  /**
+   * Determine whether or not a given entity is domain-enabled.
+   *
+   * @return string[]
+   *   List of entities supporting domain access.
+   */
+  protected function entityHasDomainField(string $entity_type, string $field = DOMAIN_ACCESS_FIELD) {
+    // Try to avoid repeated calls to getFieldMap() assuming it's expensive.
+    $this->ensureEntityFieldMap();
+    return array_key_exists($field, $this->entity_field_map[$entity_type]);
+  }
+
+  /**
+   * Ensure the local entity field map has been defined.
+   *
+   * @return string[]
+   *   List of entities supporting domain access.
+   */
+  protected function ensureEntityFieldMap() {
+    // Try to avoid repeated calls to getFieldMap() assuming it's expensive.
+    if (empty($this->entity_field_map)) {
+      $entity_manager         = \Drupal::entityManager();
+      $this->entity_field_map = $entity_manager->getFieldMap();
+    }
+  }
+
+  /**
+   * Enumerate entities of the supplied type and domain.
```

Comment:
I really don't understand what `enumerate` means in this context. The method name needs some clarification.

Reply:
My intention is to return a list of the entity instances for a given condition. So for 'node' type then return a list of node ids for the supplied domain.
I am not clear where the issue lies? Could you help by saying what you thought it might mean?